gk/fib.c

```diff
@@ -1901,24 +1883,17 @@ list_hash_table_neighbors(lua_State *l, enum gk_fib_action action,
 	uint32_t next = 0;
 	const void *key;
 	void *data;
-	int32_t index;
-
-	static bool assigned_type_neighbor_dentry = false;
-	static uint32_t correct_ctypeid_neighbor_dentry;
 
 	void *cdata;
 
-	if (!assigned_type_neighbor_dentry) {
-		correct_ctypeid_neighbor_dentry = luaL_get_ctypeid(l,
-			CTYPE_STRUCT_NEIGHBOR_DUMP_ENTRY_PTR);
-		assigned_type_neighbor_dentry = true;
-	}
+	uint32_t correct_ctypeid_neighbor_dentry = luaL_get_ctypeid(l,
+		CTYPE_STRUCT_NEIGHBOR_DUMP_ENTRY_PTR);
 
-	index = rte_hash_iterate(neigh_ht->hash_table,
+	int32_t index = rte_hash_iterate(neigh_ht->hash_table,
 		(void *)&key, &data, &next);
 	while (index >= 0) {
 		struct gk_neighbor_dump_entry dentry;
-		struct ether_cache *eth_cache = &neigh_ht->cache_tbl[index];
+		struct ether_cache *eth_cache = data;
```

Comment:
Having a small patch to address this bug is better instead of conflating two unrelated bugs in the same patch. It's not an issue for the review, but it'll make later inspection of `git log` a lot easier.